### PR TITLE
[Develop] Split mina_graphql.ml to a set of files

### PR DESCRIPTION
**Pull request by georgeee**
Merge to `develop` of #13956

### CODE REVIEW

**Review by MartinMinkov — APPROVED** [review]

**Review by psteckler — APPROVED** [review]

**Comment by georgeee** [comment]
!ci-build-me

File: src/app/rosetta/lib/block.ml
Comment (Sventimir) [inline src/app/rosetta/lib/block.ml:1]: These changes are already merged into `berkeley` and will find their way to `develop` eventually. They don't seem relevant to this PR at all, so I wonder why do you include them?

File: src/lib/mina_graphql/itn_zkapps.ml [hunk 1]
```diff
@@ -0,0 +1,293 @@
...
+let is_zkapp_deployed kp ledger =
+  match
+    Option.try_with (fun () ->
+        let account = Utils.account_of_kp kp ledger in
+        Option.is_some account.zkapp )
+  with
+  | Some true ->
+      true
+  | _ ->
+      false
```
Comment (Sventimir) [inline src/lib/mina_graphql/itn_zkapps.ml:100]:
```suggestion
Option.try_with (fun () ->
    let account = Utils.account_of_kp kp ledger in
    Option.is_some account.zkapp )
|> Option.value ~default:false
```

File: src/lib/mina_graphql/itn_zkapps.ml [hunk 2]
```diff
@@ -0,0 +1,293 @@
+open Core
```
Comment (Sventimir) [inline src/lib/mina_graphql/itn_zkapps.ml:1]: Why `itn_zkapps`? How are they different to usual zkapps?

File: src/lib/mina_graphql/types.ml [hunk 3]
```diff
@@ -0,0 +1,3216 @@
+open Core
+open Async
+open Mina_base
+open Mina_transaction
+module Ledger = Mina_ledger.Ledger
+open Signature_lib
+open Currency
+open Schema
+
+include struct
```
Comment (Sventimir) [inline src/lib/mina_graphql/types.ml:10]: Is this done to narrow the scope of opening `Graphql_lib.Scalars`? If so, please explain this in a comment, because it looks confusing.

**Review by Sventimir — APPROVED** [review]
I very much like the idea of splitting large modules. I added a couple of minor comments, but overall it looks good to me!

File: src/lib/mina_graphql/itn_zkapps.ml (hunk 2 above)
Comment (georgeee) [reply src/lib/mina_graphql/itn_zkapps.ml:1]: In `zkapps` module regular logic to send a zkapp command is kept. In `itn_zkapps` there is a code for scheduler of a series of zkapp commands designed for testing on ITN. This is quite different and more complex, also not to be used in production

File: src/app/rosetta/lib/block.ml
Comment (georgeee) [reply src/app/rosetta/lib/block.ml:1]: I just merged my branch raised against `berkeley` to `develop`, as this is the easiest way. This is a merging PR< not the main PR

File: src/lib/mina_graphql/types.ml (hunk 3 above)
Comment (georgeee) [reply src/lib/mina_graphql/types.ml:10]: Yes, I'll rewrite it slightly to remove the `include struct`

File: src/lib/mina_graphql/itn_zkapps.ml (hunk 1 above)
Comment (georgeee) [reply src/lib/mina_graphql/itn_zkapps.ml:100]: I rewrote it to:

```ocaml
try Option.is_some (Utils.account_of_kp kp ledger).zkapp with _ -> false
```

File: src/lib/mina_graphql/types.ml (hunk 3 above)
Comment (georgeee) [reply src/lib/mina_graphql/types.ml]: See commit [a7b61d2](https://github.com/MinaProtocol/mina/pull/14037/commits/a7b61d2c9e312b3994a65e4f1e7efd2869d282a6)

File: src/lib/mina_graphql/itn_zkapps.ml (hunk 1 above)
Comment (georgeee) [reply src/lib/mina_graphql/itn_zkapps.ml]: See commit [a7b61d2](https://github.com/MinaProtocol/mina/pull/14037/commits/a7b61d2c9e312b3994a65e4f1e7efd2869d282a6)

File: src/lib/mina_graphql/itn_zkapps.ml (hunk 2 above)
Comment (Sventimir) [reply src/lib/mina_graphql/itn_zkapps.ml:1]: If it's not going to be used in production, then I wonder why it is part of the repo? Why does the daemon need to be specially augmented for ITN testing? This feels wrong. After all, tests should test the production code, not some other code written specifically to support testing… Anyway, I guess this has been written previously and you're just moving it around and adapting, so I won't object to that. Still, I'd like to learn the answers to the questions above.

**Comment by georgeee** [comment]
!ci-build-me

File: src/lib/mina_graphql/types.ml (hunk 3 above)
Comment (Sventimir) [reply src/lib/mina_graphql/types.ml]: Thanks!

File: src/lib/mina_graphql/itn_zkapps.ml (hunk 2 above)
Comment (georgeee) [reply src/lib/mina_graphql/itn_zkapps.ml:1]: It's used for testnet testing, for ITN specifically. When I say it's not for production, I mean that it's not to be used on mainnet. We cut ITN-related functionality off mainnet builds by using compile-time configuration.

Comment (Sventimir) [reply src/lib/mina_graphql/itn_zkapps.ml:1]: But that's my point exactly! Effectively this means ITN is testing a different node than the one we're going to release for mainnet. What do I misunderstand here?

Comment (georgeee) [reply src/lib/mina_graphql/itn_zkapps.ml:1]: That's certainly true. That's why we try to include as little as possible into this ITN toolkit and be transparent about that. Having this functionality in mainnet is a bad idea, but not having it in test network reduces our ability to test.

ITN toolkit is a compromise in the middle.

Comment (Sventimir) [reply src/lib/mina_graphql/itn_zkapps.ml:1]: Understood. Thanks for explanation!

**Comment by georgeee** [comment]
!ci-build-me

**Comment by georgeee** [comment]
!ci-build-me
